### PR TITLE
Support seq indexing

### PR DESCRIPTION
**Pull request by tbekas**

### CODE REVIEW

**Review by markspanbroek — APPROVED** [review]
Very nice, thanks @tbekas!

File: questionable/indexing.nim [hunk 1]
```diff
@@ -10,3 +13,5 @@ macro `.?`*(expression: typed, brackets: untyped{nkBracket}): untyped =
         `expression`[`index`].some
       except KeyError:
         T.none
+      except IndexDefect:
```
[the remark this answers is not in the document]
Comment (arnetheduck) [reply questionable/indexing.nim:16]: https://nim-lang.org/docs/manual.html#definitions

> the following program is invalid; even though the code purports to catch the IndexDefect from an out-of-bounds array access, the compiler may instead choose to allow the program to die with a fatal error.

[the remark this answers is not in the document]
Comment (markspanbroek) [reply questionable/indexing.nim:16]: Good point, I missed that. I thought we were catching an `IndexError`, but indeed it's a `Defect`.

[the remark this answers is not in the document]
Comment (arnetheduck) [reply questionable/indexing.nim:16]: the canonical way to  do this would be to do a length pre-check

[the remark this answers is not in the document]
Comment (tbekas) [reply questionable/indexing.nim:16]: Thanks for the explanation, I didn't know that Defects may not be caught

**Pull request merged**